src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2618,6 +2667,32 @@ void SystemNative_GetDomainSocketSizes(int32_t* pathOffset, int32_t* pathSize, i
     *addressSize = sizeof(domainSocket);
 }
 
+int32_t SystemNative_Disconnect(intptr_t socket)
+{
+    int fd = ToFileDescriptor(socket);
+    int err;
+
+#if defined(__linux__)
+    // On Linux, we can disconnect a socket by connecting to AF_UNSPEC.
+    // For TCP sockets, this causes an abortive close.
```

Comment:
What happens if this is used on connectionless sockets?

Reply:
It will give back the local address, so it may be used by another socket. The peer doesn't observe something.